### PR TITLE
Add resource provider registry

### PR DESCRIPTION
**Pull request by edwardfeng-db**
## Changes

- Part of https://github.com/databricks/terraform-provider-databricks/pull/3320
- Add resource provider registry to avoid redundant alias and customization definitions
- Update `Library` resource to follow the ResourceProvider interface

## Tests

- [ ] `make test` run locally
- [ ] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [ ] using Go SDK

### CODE REVIEW

File: common/reflect_resource.go [hunk 1]
```diff
@@ -38,6 +38,25 @@ var kindMap = map[reflect.Kind]string{
...
+// Example:
+//
+//	func init() {
+//		 common.RegisterResourceProvider(jobs.JobSettings{}) = JobSettingsResource{}
+//	}
+func RegisterResourceProvider(v any, r ResourceProvider) {
+	if resourceProviderRegistry == nil {
+		resourceProviderRegistry = make(map[string]ResourceProvider)
+	}
+	resourceProviderRegistry[getNameForType(reflect.ValueOf(v).Type())] = r
```
Comment (mgyucht) [inline common/reflect_resource.go:57]: nit: let's panic if the same type is registered more than once?

File: clusters/resource_library.go [hunk 2]
```diff
@@ -12,14 +12,24 @@ import (
...
-		}
-		return m
+type LibraryResource struct {
+	compute.Library
+}
+
+func (LibraryResource) Aliases() map[string]map[string]string {
+	return map[string]map[string]string{}
+}
+
```
Comment (mgyucht) [inline clusters/resource_library.go:22]: Is this still needed? I thought we made aliases optional?

**Review by mgyucht — APPROVED** [review]
Two small nits, otherwise LGTM. Very excited to see the Jobs migration.

File: clusters/resource_library.go (hunk 2 above)
Comment (edwardfeng-db) [reply clusters/resource_library.go:22]: ah, yes, good catch

**Pull request merged**